src/System.Text.Http.Parser/HttpParser.cs

```diff
@@ -23,6 +23,18 @@ public class HttpParser : IHttpParser
         private const byte ByteQuestionMark = (byte)'?';
         private const byte BytePercentage = (byte)'%';
 
+        private readonly bool _showErrorDetails;
+
+        public HttpParser()
+            : this(showErrorDetails: true)
```

Comment:
Should we default this to false?

Reply:
I think we should get to a state first where Kestrel is consuming our packages before we start fine tuning the api surface. With this change, we're just one test away from passing all their tests.